src/controllers/clientController.js

```diff
@@ -171,3 +171,41 @@ exports.getChefById = async (req, res) => {
     res.status(500).json({ error: error.message });
   }
 };
+
+exports.dishesRatings = async (req, res) => {
+  const token = req.cookies.jwt;
+  const decodedToken = decodeJwtToken(token);
+
+  const { id } = req.params;
+  const { rating } = req.body;
+
+  try {
+    const dish = await Dish.findById(id);
+
+    const existingRatingIndex = dish.ratings.findIndex((r) =>
+      r.userId.equals(decodedToken.userId)
+    );
+
+    if (existingRatingIndex !== -1) {
+      dish.ratings[existingRatingIndex].rating = rating;
+    } else {
+      const newRating = { userId: decodedToken.userId, rating };
+      dish.ratings.push(newRating);
+    }
+
+    const ratingsCount = dish.ratings.length;
+    const ratingSum = dish.ratings.reduce(
+      (sum, value) => sum + value.rating,
+      0
+    );
+    const averageRating = ratingsCount > 0 ? ratingSum / ratingsCount : 0;
+
+    dish.ratingAve = averageRating.toFixed(1);
```

Comment:
You mentioned here `dish.ratingAve` but there is no `ratingAve` in the dish Schema, so this line which is `203` is doing nothing for now. So now if you go and rate a dish, then you won't see the ratingAve.